src/scenic/syntax/veneer.py

```diff
@@ -111,1000 +132,1341 @@
+    value = {'position': RelativeTo(offset, ego()).toVector(), 'parentOrientation': ego().orientation}
+    return Specifier("OffsetBy", {'position': 1, 'parentOrientation': 3}, value)
 
-	Specifies :prop:`position`, with no dependencies.
+def OffsetAlongSpec(direction, offset):
+    """The :specifier:`offset along {X} by {Y}` polymorphic specifier.
 
-	This uses the given object's :prop:`visibleRegion` property, and so correctly
-	handles the view regions of Points, OrientedPoints, and Objects.
-	"""
-	if not isinstance(base, Point):
-		raise RuntimeParseError('specifier "visible from O" with O not a Point')
-	return Specifier('position', Region.uniformPointIn(base.visibleRegion))
+    Specifies :prop:`position` and :prop:`parentOrientation`, with no dependencies.
 
-def VisibleSpec():
-	"""The :specifier:`visible` specifier (equivalent to :specifier:`visible from ego`).
+    Allowed forms::
 
-	Specifies :prop:`position`, with no dependencies.
-	"""
-	return VisibleFrom(ego())
+        offset along <heading> by <vector>
+        offset along <field> by <vector>
+    """
+    pos = OffsetAlong(ego(), direction, offset)
+    parentOrientation = ego().orientation
+    return Specifier("OffsetAlong", {'position': 1, 'parentOrientation': 3},  {'position': pos, 'parentOrientation': parentOrientation})
 
-def NotVisibleFrom(base):
-	"""The :grammar:`not visible from <Point>` specifier.
-
-	Specifies :prop:`position`, depending on :prop:`regionContainedIn`.
-
-	See `VisibleFrom`.
-	"""
-	if not isinstance(base, Point):
-		raise RuntimeParseError('specifier "not visible from O" with O not a Point')
-	def helper(self):
-		region = self.regionContainedIn
-		if region is None:
-			if currentScenario._workspace is None:
-				raise RuntimeParseError('"not visible" specifier with no workspace defined')
-			region = currentScenario._workspace.region
-		return Region.uniformPointIn(region.difference(base.visibleRegion))
-	return Specifier('position', DelayedArgument({'regionContainedIn'}, helper))
+def Beyond(pos, offset, fromPt=None):
+    """The :specifier:`beyond {X} by {Y} from {Z}` polymorphic specifier.
 
-def NotVisibleSpec():
-	"""The :specifier:`not visible` specifier (equivalent to :specifier:`not visible from ego`).
+    Specifies :prop:`position` and :prop:`parentOrientation`, with no dependencies.
 
-	Specifies :prop:`position`, depending on :prop:`regionContainedIn`.
-	"""
-	return NotVisibleFrom(ego())
+    Allowed forms::
 
-def OffsetBy(offset):
-	"""The :grammar:`offset by <vector>` specifier.
+        beyond <vector> by <number> [from <vector>]
+        beyond <vector> by <vector> [from <vector>]
 
-	Specifies :prop:`position`, with no dependencies.
-	"""
-	offset = toVector(offset, 'specifier "offset by X" with X not a vector')
-	pos = RelativeTo(offset, ego()).toVector()
-	return Specifier('position', pos)
+    If the :grammar:`from <vector>` is omitted, the position of ego is used.
+    """
+    # Ensure X can be coaxed into  vector form
```

Comment:
There are a few places where you use "coax"; let's stick with "coerce", which is the term used consistently elsewhere.